consul-lambda-registrator/consul-lambda-extension/main.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/hashicorp/go-hclog"
+	"github.com/kelseyhightower/envconfig"
+
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/client"
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/extension"
+	"github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/trace"
+)
+
+const (
+	defaultLogLevel = "info"
+	extensionName   = "consul-lambda-extension"
+)
+
+func main() {
+	logger := hclog.New(&hclog.LoggerOptions{
+		Level: hclog.LevelFromString(getEnvOrDefault("LOG_LEVEL", defaultLogLevel)),
+	})
+
+	trace.SetLogger(trace.HCLog{Logger: logger})
+	trace.Enabled(strings.ToLower(getEnvOrDefault("TRACE_ENABLED", "false")) == "true")
+
+	err := realMain(logger.Named(extensionName))
+	if err != nil {
+		logger.Error("fatal error, exiting", "error", err)
+		os.Exit(1)
+	}
+}
+
+func realMain(logger hclog.Logger) error {
+	trace.Enter()
+	defer trace.Exit()
+
+	cfg, err := configure()
+	if err != nil {
+		return err
+	}
+
+	cfg.Logger = logger
+	ext := extension.NewExtension(cfg)
+
+	// Handle interrupts and shutdown notification
+	ctx, cancel := context.WithCancel(context.Background())
+	shutdownChannel := make(chan struct{})
+	go func() {
+		interruptChannel := make(chan os.Signal, 1)
+		signal.Notify(interruptChannel, syscall.SIGTERM, syscall.SIGINT)
+
+		select {
+		case s := <-interruptChannel:
+			logger.Info("Received signal, exiting", "signal", s)
+		case <-shutdownChannel:
+			logger.Info("Received shutdown event, exiting")
+		}
+		// Cancel our context so that all servers and go-routines exit gracefully.
+		cancel()
+	}()
+
+	err = ext.Serve(ctx)
+	if err != nil {
+		logger.Error("processing failed with an error", "error", err)
+	}
+
+	// Signal that it's time to shutdown when the extension returns.
+	shutdownChannel <- struct{}{}
+
+	return err
+}
+
+func configure() (*extension.Config, error) {
+	trace.Enter()
+	defer trace.Exit()
+
+	// Load the configuration from the environment.
+	cfg := &extension.Config{}
+	err := envconfig.Process("consul", cfg)
+	if err != nil {
+		return cfg, fmt.Errorf("failed to load configuration from environment: %w", err)
+	}
+
+	if cfg.ServiceName == "" {
+		// If the service name wasn't explicitly configured then default to the Lambda function's name.
+		cfg.ServiceName = os.Getenv("AWS_LAMBDA_FUNCTION_NAME")
```

Comment:
Just talking through the safety behind this:
* This name doesn't get used to register services into Consul so it can't squash other services
* It get used to fetch data from param store. This is fine because it needs the IAM permissions to fetch the data.

When would users want to set this?

Reply:
Yes, I don't think that we want users to be allowed to change this.. This is related to your comment on the other PR. I think we should probably remove the `CONSUL_SERVICE_NAME` configuration.. if a user sets this to something other than the function name it will break things because the Lambda registrator is pushing data to param store using `/CONSUL_EXTENSION_DATA_PATH/<ap>/<ns>/<lambda-function-name>`

Reply:
I've removed the ability to make `CONSUL_SERVICE_NAME` configurable.. It is now always set to value of `AWS_LAMBDA_FUNCTION_NAME`